java/code/src/com/redhat/rhn/domain/formula/FormulaFactory.java

```diff
@@ -1107,4 +1109,75 @@ public static String getClusterPillarDir() {
         return dataDir + PILLAR_DIR;
     }
 
+    /**
+     * Find endpoint information from given formula data
+     * @param formulaName name of the formula to examine
+     * @param formulaData formula data to extract information from
+     * @return list of endpoint information objects
+     */
+    public static List<EndpointInfo> getEndpointsFromFormulaData(String formulaName, FormulaData formulaData) {
+        List<EndpointInfo> endpointInfos = new ArrayList<>();
+        endpointInfos.addAll(getExportersEndpoints(formulaData));
+        endpointInfos.addAll(getApplicationEndpoints(formulaName, formulaData));
+        return endpointInfos;
+    }
+
+    private static List<EndpointInfo> getApplicationEndpoints(String formulaName, FormulaData formulaData) {
+        // TODO
+        return Collections.emptyList();
+    }
```

Comment:
I would remove this method as of today, and add it only when we will actually support application endpoints.